planner/core/expression_rewriter.go

```diff
@@ -886,6 +886,12 @@ func (er *expressionRewriter) Leave(originInNode ast.Node) (retNode ast.Node, ok
 			return retNode, false
 		}
 
+		// check the maxDecimalScale and maxDecimalWidth
+		er.err = er.checkDecimal(v.Tp, arg.String())
```

Comment:
should we also update `expression/simple_rewriter.go`?

Reply:
I don't think we need to update expression/simple_rewrite.go. For the first reason, a similar check function called [`checkTimePrecision`](https://github.com/pingcap/tidb/blob/master/planner/core/expression_rewriter.go#L884) in expression_rewriter did not appear in simple_rewriter. Besides, I think the simple_rewriter works when we use the table partition. In MySQL, a partitioning key must be either an integer column or an expression that resolves to an integer. So, I do not think there have a chance to use cast(someType, decimal).